langchain/retrievers/weaviate_hybrid_search.py

```diff
@@ -40,6 +40,18 @@ def __init__(
         if attributes is not None:
             self._query_attrs.extend(attributes)
 
+        self._create_schema()
```

Comment:
Ideally, initialization code is fast and does not include any side-effects (i.e., not connecting to outside databases or creating schemas).

Would it make sense to expose this as a public method that the user can invoke if they need it?

Reply:
> Would it make sense to expose this as a public method that the user can invoke if they need it?

No. Given the way the class has been designed, schema creation has to happen before they upload docs into the database. I'd rather make it implicit than count on the user to make the call themselves.

Reply:
What do you think about adding a parameter that allows a user to control this behavior in the init?

The reason I'm concerned is that this pattern make a network request on every initialization of the retriever (checking if schema exists). This will get in the way of any users trying to move this retriever into production. The user's code may involve may involve single instantiation of the weaviate client, but instantiations of the retriever with the assumption that there isn't any over-head.

Reply:
> What do you think about adding a parameter that allows a user to control this behavior in the init?

This is a good idea if the user already has an identical schema inside weaviate. If not, a schema has to be created anyway because we need to tell weaviate which vectorizer to use.

so, assuming the param is called create_schema, is this how you'd see the user's code could look like?

```python
x = WeaviateHybridSearchRetriever(client, index_name="Foo", text_key="text")
y = WeaviateHybridSearchRetriever(client, index_name="Bar", text_key="text")
z = WeaviateHybridSearchRetriever(client, index_name="Foo", text_key="text", create_schema=False)

```

Reply:
Yeah usage pattern looks perfect.  I don't know if there's a better variable name, so the user knows that the schema does not get over-written ever. `create_schema_if_missing` is explicit, but maybe a bit too annoying to type?

Reply:
i think `create_schema_if_missing` is good. clarity > brevity in var names. Also, user can use intellisense to save some typing :)